### DISCUSSION TITLE
Upgrading from WCM Core Components 2.18.0 to 2.22.0 - Carousel Component Issue

### DISCUSSION THREAD

**Discussion by ar80001-BRONGA** [adobe/aem-core-wcm-components#2484] [Q&A]
Hello everyone,

We're on AEM 6.5.16 and are attempting to upgrade WCM Core Components from version 2.18.0 to 2.22.0, and I'm now receiving the following error from our custom carousel component (which extends core/wcm/components/carousel/v1/carousel) when clicking on the panel selector icon:

Uncaught Error: Syntax error, unrecognized expression: # .cmp-carousel__item--active
    at Sizzle.error (jquery.js:1503:8)
    at Sizzle.tokenize (jquery.js:2160:11)
    at Sizzle.compile (jquery.js:2548:12)
    at Sizzle.select (jquery.js:2639:16)
    at Sizzle (jquery.js:904:9)
    at Sizzle.matchesSelector (jquery.js:1470:9)
    at jQuery.filter (jquery.js:2803:15)
    at winnow (jquery.js:2787:22)
    at jQuery.fn.init.is (jquery.js:2842:12)
    at HTMLDivElement.<anonymous> (page.770458083c8db183b522c837a2474474.js:339:109)

Has anyone else encountered this error, and if you have, what did you do to fix it?

Thanks!
Art

**Comment by ar80001-BRONGA** [comment]
From doing some research this appears to be the code that's throwing the exception, and I suspect it's treating the cmp-carousel__item--active class as if it were an ID?

```
getActiveIndex: function() {
    var that = this;
    var activeIndex = 0;
    if (that._config.el && that._config.panelContainerType) {
        var items = $(this._config.el).find(that._config.panelContainerType.itemSelector);
        items.each(function(index) {
            if ($(this).is("#" + that._config.el.id + " " + that._config.panelContainerType.itemActiveSelector)) {
                activeIndex = index;
                return false
            }
        })
    }
    return activeIndex
},
```

**Reply by gbanez** [reply]
@ar80001-BRONGA  

Yes, seeing this issue as well. Are you still encountering this?

Seems there is a possible fix in 2.22.12 patch in #2513 . It gets rid of the error, but doesnt actually allow the panels to switch at least in my view.
